test/cases/DBVersioner.test.ts

```diff
@@ -0,0 +1,236 @@
+import assert from "node:assert";
+import {
+  DBVersioner,
+  MongoDBAdapter,
+  VERSION_TYPES,
+} from "../../src";
+import { loggerMock } from "../mocks/logger";
+import { dbMock, MongoClientMock } from "../mocks/mongo";
+
+
+describe("DB versioner", function () {
+  this.timeout(5000);
+
+  let mongoAdapter : MongoDBAdapter;
+  let versioner : DBVersioner;
+
+  const contractsVersion = "1.7.9";
+  const dbVersion = "109381236293746234";
+
+  before(async () => {
+
+    versioner = new DBVersioner({
+      dbVersion,
+      contractsVersion,
+      archive: false,
+      logger: loggerMock,
+    });
+
+    mongoAdapter = new MongoDBAdapter({
+      logger: loggerMock,
+      dbUri: "mongodb://mockedMongo",
+      dbName: "TestDatabase",
+      mongoClientClass: MongoClientMock,
+      versionerClass: DBVersioner,
+      dbVersion,
+      contractsVersion,
+    });
+
+    await mongoAdapter.initialize(dbVersion);
+  });
+
+  it("Should return existing temp version when calls configureVersioning()", async () => {
```

Comment:
not sure what this is testing

Reply:
It shows that DB Versioner can find correct final version if it exists
Tell me if I'm wrong